colcon_ros/package_identification/ignore.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2016-2018 Dirk Thomas
+# Licensed under the Apache License, Version 2.0
+
+from colcon_core.package_identification \
+    import PackageIdentificationExtensionPoint
+from colcon_core.package_identification import SkipLocationException
+from colcon_core.plugin_system import satisfies_version
+
+
+class IgnorePackageIdentification(PackageIdentificationExtensionPoint):
+    """
+    Ignore paths containing a `_order_isolated_packages.py` file.
+
+    Workspaces built with `ament_tools` don't contain any IGNORE marker files
+    in the install space.
+    To avoid packages being found in that directory by the discovery
+    extensions using default values it is being ignored by this extension.
+    """
+
+    # the priority needs to be higher than all other extensions
+    PRIORITY = 1000
+
+    def __init__(self):  # noqa: D107
```

Comment:
Contributor's question:
Is the end goal to add documentation to all functions ?
or should we add `noqa: DXXX` on all "trivial" function declarations when we contribute to this project ?

Reply:
All API should have a docblock.

In this case the function is implementing the interface of a the parent class. So I don't think it needs a custom documentation. The parent API should document what the function does / is supposed to do, arguments, return values, exceptions, etc. If the function does something "special" it could be documented though.